### PR TITLE
Enable --no-daemon flag for some cli tools

### LINKED ISSUE #511: --no-daemon is not valid
## Bug report

**Required Info:**

- Operating System:
  - Ubuntu 20.04
- Installation type:
  - source
- Version or commit hash:
  - bd79e43d1273b3055cb2b431821026b44c696832
- DDS implementation:
  - fast-rtps
- Client library (if applicable):
  N/A

#### Steps to reproduce issue
```
ros2 topic list --no_daemon
```
#### Expected behavior
Should be an acceptable flag, used by strategy.py
#### Actual behavior
```
$ ros2 topic list --no_daemon
usage: ros2 [-h] Call `ros2 <command> -h` for more detailed usage. ...
ros2: error: unrecognized arguments: --no_daemon
```
#### Additional information
Likely related to the recent commit by @hidmic. This one's easy enough, I have a pull request incomming.

### PR DESCRIPTION
**Pull request by dawonn-haval**
fixes #511
added daemon support to the other topic tools, except pub.

Signed-off-by: Dereck Wonnacott <dereck.wonnacott@havalus.com>

### CODE REVIEW

**Comment by dawonn-haval** [comment]
As far as I can tell, the unit tests that are failing run fine on my systems, and shouldn't have been affected by this commit.

**Comment by sloretz** [comment]
@dawonn-haval friendly ping, what do you think of @hidmic's feedback?

**Comment by dawonn-haval** [comment]
I'll push the requested changes momentarily.

**Comment by hidmic** [comment]
CI up to `ros2topic`:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/11217/)

**Comment by hidmic** [comment]
@dawonn-haval CI reports some linter errors, mind to take a look?

**Comment by dawonn-haval** [comment]
Rebase Complete.

[Build Status](http://build.ros2.org/job/Rpr__ros2cli__ubuntu_focal_amd64/46/)

**Comment by hidmic** [comment]
@dawonn-haval CI still reports linter issues.

**Comment by dawonn-haval** [comment]
https://ci.ros2.org/job/ci_linux/11446/testReport/junit/ros2topic.ros2topic.test/test_flake8/test_flake8/

These don't show up on my local system or the pull request CI check; I did what the error told me to do and pushed the change,  so I hope they are fixed now... Can you run that CI suite again please?

**Comment by dawonn-haval** [comment]
Apparently didn't add the modified files to the commit... sorry, but can you roll it once more?

**Comment by hidmic** [comment]
Alright, finally green. Thank you for your patience @dawonn-haval ! Merging.

File: ros2topic/ros2topic/verb/echo.py [hunk 1]
```diff
@@ -136,7 +142,8 @@ def cb(msg):
 def subscriber_cb_csv(truncate_length, noarr, nostr):
     def cb(msg):
         nonlocal truncate_length, noarr, nostr
-        print(message_to_csv(msg, truncate_length=truncate_length, no_arr=noarr, no_str=nostr))
+        print(message_to_csv(msg, truncate_length=truncate_length,
+                             no_arr=noarr, no_str=nostr))
```
Comment (jacobperron) [inline ros2topic/ros2topic/verb/echo.py:146]: nitpick: For future, avoid changes unrelated to the PR. It helps keep the diff minimal, reducing chance for conflict with other changes (I ran into a conflict here).

Regarding style, this type of indentation is against our [style guide](https://index.ros.org/doc/ros2/Contributing/Code-Style-Language-Versions/#python) that prefers [hanging indentation](https://www.python.org/dev/peps/pep-0008/#indentation); although our linters don't catch it so I cannot blame anyone :man_shrugging:

**Pull request merged**